src/utils/abi/index.ts

```diff
@@ -0,0 +1,25 @@
+import * as fs from "fs/promises";
+import { ContractInterface } from "ethers";
+import { isError } from "../../typeguards";
+
+/**
+ * @dev To retrieve a new ABI, the following is useful:
+ * CONTRACT_ADDRESS="0x1234..."
+ * URL="http://api.etherscan.io/api?module=contract&action=getabi&format=raw&address=${CONTRACT_ADDRESS}"
+ * curl -s "${URL}" | python -m json.tool > "./src/utils/abi/contracts/new-contract.json"
+ */
+
+/**
+ * @notice Retrieve an ABI desription from the set of known contracts.
+ * @param contractName Name of the contract ABI to retrieve.
+ * @returns Contract ABI as an ethers ContractInterface type.
+ */
+export async function getABI(contractName: string): Promise<ContractInterface> {
```

Comment:
Do you think we should be mindful of casing here? Like: `getABI("votingv2")` vs `getABI("VotingV2")`

Reply:
I'd prefer to avoid trying to normalise case here - since this just does an fs lookup we're totally dependent on how the files are named when they're stored. I think it'll work fine  as-is :+1: